client/blocks/gdpr-banner/index.jsx

```diff
@@ -29,83 +29,87 @@ const SIX_MONTHS = 6 * 30 * 24 * 60 * 60;
 
 const hasDocument = typeof document !== 'undefined';
 
-class GdprBanner extends Component {
-	static propTypes = {
-		recordCookieBannerOk: PropTypes.func,
-		recordCookieBannerView: PropTypes.func,
-		translate: PropTypes.func,
-	};
+function shouldShowBanner() {
+	// Don't render banner in SSR.
+	if ( ! hasDocument ) {
+		return false;
+	}
+	const cookies = cookie.parse( document.cookie );
+	if ( cookies.sensitive_pixel_option === 'yes' || cookies.sensitive_pixel_option === 'no' ) {
+		return false;
+	}
+	if ( isWpMobileApp() ) {
+		return false;
+	}
+	if ( isCurrentUserMaybeInGdprZone() ) {
+		return true;
+	}
+	return false;
+}
 
-	static defaultProps = {
-		recordCookieBannerOk: noop,
-		recordCookieBannerView: noop,
-		translate: identity,
-	};
+function GdprBanner( props ) {
+	const [ hideBanner, setHideBanner ] = useState( false );
+	const [ renderBanner, setRenderBanner ] = useState( false );
```

Comment:
Can we unify both state bits into one?
```jsx
function GdprBanner() {
  const [ showBanner, setShowBanner ] = useState( false );

  useEffect( () => {
    setShowBanner( shouldShowBanner() );
  }, [] );

  if ( ! showBanner ) {
    return null;
  }

  return <TheRealGdprBanner />;
}
```

Reply:
We could, sure. I assumed there was a good reason for keeping it in the DOM and just hiding it through CSS? Pinging @MicBosi and @lsinger in case they know something.

Reply:
@jsnajdr There appears to be an animation, which we probably want to preserve.

Reply:
Should we instead flip it around and always hide in CSS?

@MicBosi / @lsinger : would there be any issue in doing that?

Reply:
> There appears to be an animation, which we probably want to preserve.

Then it might be clearer to use one state variable with three states:
- `not-rendered`: the state on initial render
- `rendered`: transition to this on second render if we want to show the banner
- `rendered-but-hidden`: transition to this after clicking "Got it!", hiding is animated

Reply:
Done.